files/000-default.conf

```diff
@@ -0,0 +1,31 @@
+<VirtualHost *:80>
+	# The ServerName directive sets the request scheme, hostname and port that
+	# the server uses to identify itself. This is used when creating
+	# redirection URLs. In the context of virtual hosts, the ServerName
+	# specifies what hostname must appear in the request's Host: header to
+	# match this virtual host. For the default virtual host (this file) this
+	# value is not decisive as it is used as a last resort host regardless.
+	# However, you must set it for any further virtual host explicitly.
+	#ServerName www.example.com
+
+	ServerAdmin webmaster@localhost
+	DocumentRoot /var/www/html
+
+	# Available loglevels: trace8, ..., trace1, debug, info, notice, warn,
+	# error, crit, alert, emerg.
+	# It is also possible to configure the loglevel for particular
+	# modules, e.g.
+	#LogLevel info ssl:warn
+
+	ErrorLog "|$ tee {APACHE_LOG_DIR}/error.log"
+    CustomLog "|$ tee ${APACHE_LOG_DIR}/access.log" combined
```

Comment:
Apache logs are piped through tee and log files to enable both container log output and file append for loki.

Reply:
Isn't repeating the `CustomLog` creating multiple log stream? To quote the [apache document](https://httpd.apache.org/docs/2.4/mod/mod_log_config.html):

> Three directives are provided by this module: [TransferLog](https://httpd.apache.org/docs/2.4/mod/mod_log_config.html#transferlog) to create a log file, [LogFormat](https://httpd.apache.org/docs/2.4/mod/mod_log_config.html#logformat) to set a custom format, and [CustomLog](https://httpd.apache.org/docs/2.4/mod/mod_log_config.html#customlog) to define a log file and format in one step. The TransferLog and CustomLog directives can be used multiple times in each server to cause each request to be logged to multiple files.

So this could be update to something like this, maybe?

```
CustomLog "${APACHE_LOG_DIR}/access.log" combined
CustomLog /dev/stdout combined
```

Reply:
I thought pipe operation would be consistent with ErrorLog but it does seem better to use native log stream directives. Updated, thank you for the comment!